lib/telemetry/helper.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Telemetry.Helper do
+  @moduledoc """
+  Helper functions for seeing all telemetry events.
+
+  Only for use in development.
+  """
+
+  require Logger
+
+  @doc """
+  Attach to all telemetry events and log them.
+
+  You can pass in a filter so that you only see telemetry events for a specific module.
+  E.g., if you want to see all telemetry events for Phoenix you can pass in `[:phoenix]`.
+  For more specific events you can pass in more specific names like `[:phoenix, :router_dispatch]` or `[:phoenix, :router_dispatch, :stop]`.
+  """
+  def attach(filter \\ nil) do
+    # Start the tracer
+    :dbg.start()
+
+    # Create tracer process with a function that pattern matches out the three arguments the telemetry calls are made with.
+    :dbg.tracer(
+      :process,
+      {
+        fn
+          {_, _, _, {_mod, :execute, [name, measurement, metadata]}}, _state ->
+            handler(name, metadata, measurement, filter)
+        end,
+        nil
+      }
+    )
+
+    # Trace all processes
+    :dbg.p(:all, :c)
+
+    # Trace calls to the functions used to emit telemetry events
+    :dbg.tp(:telemetry, :execute, 3, [])
+  end
+
+  @doc """
+  Stop the tracer and therefore stop listening for telemetry events.
+  """
+  def stop do
+    :dbg.stop()
+  end
+
+  defp handler(name, metadata, measure, filter) do
+    unless filtered?(name, filter) do
+      # credo:disable-for-next-line Credo.Check.Warning.IoInspect
+      IO.inspect(%{name: name, metadata: metadata, measure: measure})
```

Comment:
Why is this an `IO.inspect`?

Reply:
This is made to be used in `iex` and `IO.inspect` is cleaner to read than a log. Also, with a log you have to choose a log level. So, if we chose `Logger.info` and the app is set to `:notice` then you wouldn't see the log, etc. etc.